packages/compiler-cli/test/ngtsc/incremental_error_spec.ts

```diff
@@ -0,0 +1,297 @@
+
+      // Clear the error and trigger the rebuild.
+      env.write('test.ts', `
+        import {Component} from '@angular/core';
+
+        @Component({selector: 'test-cmp', template: '...'})
+        export class TestCmp {}
+      `);
+
+      env.flushWrittenFileTracking();
+      env.driveMain();
+
+      const written = env.getFilesWrittenSinceLastFlush();
+      expectToHaveWritten([
+        // The file which had the error should have been emitted, of course.
+        '/test.js',
+
+        // Because TestCmp belongs to a module, the module's file should also have been
+        // re-emitted.
+        '/module.js',
+
+        // Because TargetCmp also belongs to the same module, it should be re-emitted since
+        // TestCmp's elector may have changed.
+        '/target.js',
+      ]);
+    });
+
+    it('should recover from an error even across multiple NgModules', () => {
+      // This test is a variation on the above. Two components (CmpA and CmpB) exist in an NgModule,
+      // which indirectly imports a LibModule (via another NgModule in the middle). The test is
+      // designed to verify that CmpA and CmpB are re-emitted if somewhere upstream in the NgModule
+      // graph, an error is fixed. To check this, LibModule is broken and then fixed in incremental
+      // build steps.
+      env.write('a.ts', `
+        import {Component} from '@angular/core';
+
+        @Component({selector: 'test-cmp', template: '...'})
+        export class CmpA {}
+      `);
+      env.write('b.ts', `
+        import {Component} from '@angular/core';
+
+        @Component({selector: 'target-cmp', template: '...'})
+        export class CmpB {}
+      `);
+      env.write('module.ts', `
+        import {NgModule} from '@angular/core';
+        import {LibModule} from './lib';
+        import {CmpA} from './a';
+        import {CmpB} from './b';
+
+        @NgModule({
+          imports: [LibModule],
+          exports: [LibModule],
+        })
+        export class IndirectModule {}
+
+        @NgModule({
+          declarations: [CmpA, CmpB],
+          imports: [IndirectModule],
+        })
+        export class Module {}
+      `);
+      env.write('lib.ts', `
+        import {Component, NgModule} from '@angular/core';
+
+        @Component({
+          selector: 'lib-cmp',
+          template: '...',
+        })
+        export class LibCmp {}
+
+        @NgModule({
+          declarations: [LibCmp],
+          exports: [LibCmp],
+        })
+        export class LibModule {}
+      `);
+
+      // Start with a clean compilation.
+      env.driveMain();
+
+      // Introduce the error in LibModule
+      env.write('lib.ts', `
+      import {Component, NgModule} from '@angular/core';
+
+      @Component({
+        selector: 'lib-cmp',
+        template: '...',
+      })
+      export class LibCmp {}
+
+      @NgModule({
+        declarations: [LibCmp],
+        exports: [LibCmp],
+      })
+      export class LibModule
```

Comment:
I was looking for the error for a while, so I'd add a comment:

```suggestion
      export class LibModule // missing braces
```